operator/internal/central/extensions/reconcile_tls_ca_rotation_test.go

```diff
@@ -0,0 +1,196 @@
+package extensions
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"testing"
+	"time"
+
+	"github.com/cloudflare/cfssl/helpers"
+	"github.com/go-logr/logr"
+	"github.com/stackrox/rox/generated/storage"
+	platform "github.com/stackrox/rox/operator/api/v1alpha1"
+	commonExtensions "github.com/stackrox/rox/operator/internal/common/extensions"
+	"github.com/stackrox/rox/operator/internal/types"
+	"github.com/stackrox/rox/operator/internal/utils"
+	"github.com/stackrox/rox/operator/internal/utils/testutils"
+	"github.com/stackrox/rox/pkg/certgen"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stretchr/testify/require"
+	coreV1 "k8s.io/api/core/v1"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestCentralCARotation(t *testing.T) {
+	t.Setenv(envCentralCARotationEnabled, "true")
+	baseCase := secretReconciliationTestCase{
+		Spec: basicSpecWithScanner(true, true),
+	}
+	central := buildFakeCentral(baseCase)
+	client := buildFakeClient(t, baseCase, central)
+
+	// wrap reconcileCentralTLS so we can inject a custom currentTime
+	runAt := func(ctx context.Context, central *platform.Central, c ctrlClient.Client, d ctrlClient.Reader, statusUpdater func(updateStatusFunc), log logr.Logger, at time.Time) error {
+		certificateNotBefore := at.Add(-5 * time.Minute)
```

Comment:
Please comment on the significance of the 5 minutes, is it related to the padding that I recall we have somewhere around validity, or is this just "some non-zero time"?

Reply:
I was mimicking what the cfssl library does, presumably to prevent issues due to clock drift. On a test running on a single machine this shouldn't matter, so I might as well remove this.

Reply:
Removed the 5 minute backdating in the tests.